WooCommerce/Classes/ViewRelated/Blaze/AdDestination/BlazeAdDestinationSettingViewModel.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+
+final class BlazeAdDestinationSettingViewModel: ObservableObject {
+    enum DestinationURLType {
+        case product
+        case home
+    }
+
+    var productURL: String
+    var homeURL: String
+
+    @Published var selectedDestinationType: DestinationURLType
+
+    @Published var parameters: [BlazeAdURLParameter]
+
+    // Text to be shown on the view for remaining available characters for custom added parameters.
+    var remainingCharactersLabel: String {
+        let remainingCharacters = calculateRemainingCharacters()
+        let lengthText = String.pluralize(remainingCharacters,
+                                             singular: Localization.parameterCharactersLimit.singular,
+                                             plural: Localization.parameterCharactersLimit.plural)
+        return String(format: lengthText, remainingCharacters)
+    }
+
+    // Text to be shown on the view for the final ad campaign URL including parameters, if any.
+    var finalDestinationLabel: String {
+        return String(format: Localization.finalDestination, buildFinalDestinationURL())
+    }
+
+    init (productURL: String,
+          homeURL: String,
+          selectedDestinationType: DestinationURLType = .product,
+          parameters: [BlazeAdURLParameter] = [BlazeAdURLParameter(key: "key1", value: "value1")]) {
+        self.productURL = productURL
+        self.homeURL = homeURL
+        self.selectedDestinationType = selectedDestinationType
+        self.parameters = parameters
+    }
+
+    func setDestinationType(type: DestinationURLType) {
+        selectedDestinationType = type
+    }
+
+    private func buildParameterString() -> String {
+        var parameterString = ""
+        for parameter in parameters {
+            // In URL format, the parameter is written such as "key=value".
+            parameterString += parameter.key + "=" + parameter.value
+
+            // If it's not the last parameter, add an ampersand.
+            if parameter != parameters.last {
+                parameterString += "&"
+            }
+        }
+
+        return parameterString
+    }
+
+    private func buildFinalDestinationURL() -> String {
+        let baseURL: String
+        switch selectedDestinationType {
+        case .product:
+            baseURL = productURL
+        case .home:
+            baseURL = homeURL
+        }
```

Comment:
Nit: this can be moved to a property for `DestinationURLType`.

Reply:
I'm not sure how to implement that. 

I tried this:

```
    enum DestinationURLType {
        case product
        case home

        var destinationURL: String {
            switch self {
            case .product:
                return productURL
            case .home:
                return homeURL
            }
        }
    }
```

but productURL and homeURL aren't accessible from inside the enum.

Putting it as a computed var instead seems to be a good alternative:

```
    var destinationURL: String {
        switch selectedDestinationType {
        case .product:
            return productURL
        case .home:
            return homeURL
        }
    }
```

What do you think?

Reply:
Looks good 👍 I missed that you are using those properties injected to the view model.